docs/reference/commandline/login.md

```diff
@@ -47,6 +47,14 @@ to your Docker account without entering your password. For Docker Hub, the
 `--username` flag is specified. The device code flow is a secure way to sign
 in. See [Authenticate to Docker Hub using device code](#authenticate-to-docker-hub-using-device-code).
 
+### Privileged user requirement
+
+`docker login` requires you to use `sudo` or be `root`, except when:
+
+- Connecting to a remote daemon, such as a `docker-machine` provisioned `docker engine`.
+- The user is added to the `docker` group. This will impact the security of your system; the `docker` group is `root` equivalent.  See [Docker Daemon Attack Surface](https://docs.docker.com/engine/security/#docker-daemon-attack-surface) for details.
+- The Docker daemon is running as a non-root user, i.e. [rootless mode](https://docs.docker.com/engine/security/rootless/).
+
```

Comment:
I removed this section in 2f206fff3c25053f3d4ec28999db646cb6301b35 (hence the need for a rebase) because this is not specific to the login command; the sudo restriction applies to all commands.

Reply:
Thanks! I'd noticed that and meant to comment.

Reply:
Hi @dvdksn, understood and thank you for the clarification!

@laurazard Would that mean that this PR will not be needed? Since “Rootless mode” applies to all commands as well. 

Or perhaps it would be to rename this section as “Privileged user exceptions” and add the point below which is specific for `docker login`:
> Connecting to a remote daemon, such as a `docker-machine` provisioned `docker engine`.

Reply:
> Would that mean that this PR will not be needed?

That might be the case, apologies 😓 

> Or perhaps it would be to rename this section as “Privileged user exceptions” and add the point below which is specific for `docker login`:
> > Connecting to a remote daemon, such as a `docker-machine` provisioned `docker engine`.

Not sure about that – `docker login` is a bit of a misnomer – it's not doing anything with the engine*, it's merely authenticating/saving credentials for the local user's CLI for a given registry. If you run `docker login` and then try to pull an image from a private repo you have credentials for, it doesn't matter whether you use a local engine or a remote daemon, since the CLI will grab the credentials and send them on a per-pull basis to whatever daemon you're using. In effect, the daemon has no concept of "logged in" or "not logged in". That depends entirely on the client calling the engine including credentials for that operation in the call.

Reply:
@laurazard Understood and thank you for the clarification! I will be closing this PR.

Reply:
No worries! @laurazard